app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/home/HomeViewPresenter.kt

```diff
@@ -27,7 +36,13 @@ class HomeViewPresenter(view: View) {
   }
 
   fun display(state: HomeViewState) {
-    progressBar.isVisible = state is Loading
+    if (state is Refreshing || state is Loading) {
+      refreshView.isRefreshing = true
+      progressBar.isVisible = true
+    } else {
+      refreshView.isRefreshing = false
+      progressBar.isVisible = false
+    }
```

Comment:
we can optimise this with 
```kotlin
refreshView.isRefreshing = state is Loading
progressBar.isVisible = state is Loading
```

Reply:
we need to check the event comes from refreshing or loading in the start of the app

Reply:
otherwise we cant test these separately,